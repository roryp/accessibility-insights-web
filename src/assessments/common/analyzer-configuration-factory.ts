// Copyright (c) Microsoft Corporation. All rights reserved.
// Licensed under the MIT License.
import { Messages } from 'common/messages';
import { RuleAnalyzerConfiguration } from 'injected/analyzers/analyzer';
import { ScannerUtils } from 'injected/scanner-utils';

export class AnalyzerConfigurationFactory {
    public static forScanner(base: Partial<RuleAnalyzerConfiguration>): RuleAnalyzerConfiguration {
        const defaultValues: Partial<RuleAnalyzerConfiguration> = {
            resultProcessor: (scanner: ScannerUtils) => scanner.getAllCompletedInstances,
            telemetryProcessor: telemetryFactory => telemetryFactory.forAssessmentRequirementScan,
            analyzerMessageType: Messages.Assessment.AssessmentScanCompleted,
        };

        return {
            ...defaultValues,
            ...base,
        } as RuleAnalyzerConfiguration;
    }
}
