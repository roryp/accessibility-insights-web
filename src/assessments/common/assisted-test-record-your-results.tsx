// Copyright (c) Microsoft Corporation. All rights reserved.
// Licensed under the MIT License.
import * as React from 'react';

import * as Markup from '../markup';

export const AssistedTestRecordYourResults = () => (
    <li>
        Record your results:
        <ol>
            <li>
                Select <Markup.Term>Fail</Markup.Term> for any instances that do not meet the
                requirement.
            </li>
            <li>
                Otherwise, select <Markup.Term>Pass</Markup.Term>. Or, after you have marked all
                failures, select <Markup.Term>Pass unmarked instances.</Markup.Term>
            </li>
        </ol>
    </li>
);
