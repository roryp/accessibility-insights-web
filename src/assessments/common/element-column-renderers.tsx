// Copyright (c) Microsoft Corporation. All rights reserved.
// Licensed under the MIT License.
import { InstanceTableRow } from 'assessments/types/instance-table-data';
import { AssessmentInstanceDetailsColumn } from 'DetailsView/components/assessment-instance-details-column';
import * as React from 'react';

export function onRenderPathColumn(item: InstanceTableRow): JSX.Element {
    let textContent = '';
    if (item.instance.target) {
        textContent = item.instance.target.join(';');
    }

    return (
        <AssessmentInstanceDetailsColumn
            background={null}
            labelText={null}
            textContent={textContent}
            tooltipId={null}
            customClassName="not-applicable"
        />
    );
}

export function onRenderSnippetColumn(item: InstanceTableRow): JSX.Element {
    return (
        <AssessmentInstanceDetailsColumn
            background={null}
            labelText={null}
            textContent={item.instance.html}
            tooltipId={null}
            customClassName="not-applicable"
        />
    );
}
