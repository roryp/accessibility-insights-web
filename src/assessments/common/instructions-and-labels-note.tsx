// Copyright (c) Microsoft Corporation. All rights reserved.
// Licensed under the MIT License.

import * as React from 'react';

import * as Markup from '../markup';

export const InstructionsAndLabelsNotes = () => (
    <Markup.Emphasis>
        Note: For WCAG 2.1 compliance, both visible instructions and labels must be programmatically
        related to a widget. For WCAG 2.0 compliance, only visible instructions must be
        programmatically related to a widget.
    </Markup.Emphasis>
);
