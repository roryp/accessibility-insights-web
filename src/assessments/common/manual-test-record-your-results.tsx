// Copyright (c) Microsoft Corporation. All rights reserved.
// Licensed under the MIT License.
import * as React from 'react';

import * as Markup from '../markup';

export interface ManualTestRecordYourResultsProps {
    isMultipleFailurePossible: boolean;
}

export const ManualTestRecordYourResults = (props: ManualTestRecordYourResultsProps) => (
    <li>
        Record your results:
        <ol>
            <li>
                If you find {props.isMultipleFailurePossible ? 'any failures' : 'a failure'}, select{' '}
                <Markup.Term>Fail</Markup.Term>,
                {props.isMultipleFailurePossible
                    ? ' then add them as failure instances'
                    : ' then add the failure instance'}
                .
            </li>
            <li>
                Otherwise, select <Markup.Term>Pass</Markup.Term>.
            </li>
        </ol>
    </li>
);
