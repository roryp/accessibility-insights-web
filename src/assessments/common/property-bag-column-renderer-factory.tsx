// Copyright (c) Microsoft Corporation. All rights reserved.
// Licensed under the MIT License.
import { ColumnValueBag } from 'common/types/property-bag/column-value-bag';
import { AssessmentInstanceRowData } from 'DetailsView/components/assessment-instance-table';
import {
    propertyBagColumnRenderer,
    PropertyBagColumnRendererConfig,
} from './property-bag-column-renderer';

export class PropertyBagColumnRendererFactory {
    public static getRenderer<TPropertyBag extends ColumnValueBag>(
        configs: PropertyBagColumnRendererConfig<TPropertyBag>[],
    ): (item: AssessmentInstanceRowData<TPropertyBag>) => JSX.Element {
        return item => {
            return propertyBagColumnRenderer(item, configs);
        };
    }
}
