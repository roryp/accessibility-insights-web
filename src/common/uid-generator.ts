// Copyright (c) Microsoft Corporation. All rights reserved.
// Licensed under the MIT License.
import { v4 } from 'uuid';

export type UUIDGenerator = () => string;

export const generateUID: UUIDGenerator = v4;
