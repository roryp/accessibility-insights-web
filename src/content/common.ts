// Copyright (c) Microsoft Corporation. All rights reserved.
// Licensed under the MIT License.
import * as react from 'react';

import { ContentCreator } from 'views/content/content-page';
import { GuidanceTitle as gt } from 'views/content/guidance-title';
import { link } from './link';

export const create = ContentCreator(link);
export const React = react;
export const GuidanceTitle = gt;
