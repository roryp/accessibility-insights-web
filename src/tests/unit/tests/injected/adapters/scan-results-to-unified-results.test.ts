// Copyright (c) Microsoft Corporation. All rights reserved.
// Licensed under the MIT License.
import { IMock, Mock, MockBehavior, Times } from 'typemoq';

import { generateUID } from '../../../../../common/uid-generator';
import {
    convertScanResultsToNeedsReviewUnifiedResults,
    convertScanResultsToUnifiedResults,
} from '../../../../../injected/adapters/scan-results-to-unified-results';
import { RuleResult, ScanResults } from '../../../../../scanner/iruleresults';

describe('ScanResults to Unified Results Test', () => {
    let generateGuidMock: IMock<() => string>;

    beforeEach(() => {
        const guidStub = 'gguid-mock-stub';
        generateGuidMock = Mock.ofInstance(generateUID, MockBehavior.Strict);
        generateGuidMock
            .setup(ggm => ggm())
            .returns(() => guidStub)
            .verifiable(Times.atLeastOnce());
    });
    const nullIdentifiers = [null, undefined, {}];
    test.each(nullIdentifiers)(
        'convertScanResultsToUnifiedResults provides a defined UnifiedResult instance %s',
        scanResultStub => {
            const unifiedResults = convertScanResultsToUnifiedResults(
                scanResultStub as ScanResults,
                generateGuidMock.object,
            );
            expect(unifiedResults).toBeDefined();
        },
    );

    test('conversion works fine when there is no data in scanresults', () => {
        const scanResultsStub: ScanResults = createTestResultsWithNoData();
        expect(
            convertScanResultsToUnifiedResults(scanResultsStub, generateGuidMock.object),
        ).toMatchSnapshot();
    });

    test('conversion works with filled up passes and failures value in scan results', () => {
        const scanResultsStub: ScanResults = createTestResults();
        expect(
            convertScanResultsToUnifiedResults(scanResultsStub, generateGuidMock.object),
        ).toMatchSnapshot();
        generateGuidMock.verifyAll();
    });

    test.each(nullIdentifiers)(
        'convertScanResultsToNeedsReviewUnifiedResults provides a defined UnifiedResult instance %s',
        scanResultStub => {
            const unifiedResults = convertScanResultsToNeedsReviewUnifiedResults(
                scanResultStub as ScanResults,
                generateGuidMock.object,
            );
            expect(unifiedResults).toBeDefined();
        },
    );

    test('needs review conversion works fine when there is no data in scanresults', () => {
        const scanResultsStub: ScanResults = createTestResultsWithNoData();
        expect(
            convertScanResultsToNeedsReviewUnifiedResults(scanResultsStub, generateGuidMock.object),
        ).toMatchSnapshot();
    });

    test('needs review conversion works with filled up passes, failures and incomplete values in scan results', () => {
        const scanResultsStub: ScanResults = createTestResultsWithIncompletes();
        expect(
            convertScanResultsToNeedsReviewUnifiedResults(scanResultsStub, generateGuidMock.object),
        ).toMatchSnapshot();
        generateGuidMock.verifyAll();
    });

    function createTestResultsWithNoData(): ScanResults {
        return {
            passes: [],
            violations: [],
            incomplete: [],
            targetPageTitle: '',
            targetPageUrl: '',
        } as ScanResults;
    }

    const node1: AxeNodeResult = {
        any: [
            {
                id: 'test-id',
                message: 'any-test-message',
                data: {},
            },
        ],
        none: [],
        all: [],
        instanceId: 'id1',
        html: 'html1',
        target: ['target1', 'id1'],
        failureSummary: 'how to fix 1',
    };

    const node2: AxeNodeResult = {
        any: [],
        none: [
            {
                id: 'none-test-id',
                message: 'none-test-message',
                data: {},
            },
        ],
        all: [],
        instanceId: 'id2',
        snippet: 'html2',
        target: ['target2', 'id2'],
        failureSummary: 'how to fix 2',
    } as AxeNodeResult;

    const node3: AxeNodeResult = {
        any: [],
        none: [],
        all: [
            {
                id: 'all-test-id',
                message: 'all-test-message',
                data: {},
            },
        ],
        instanceId: 'id3',
        html: 'html3',
        target: ['target3', 'id3'],
        failureSummary: 'how to fix 3',
    };

    const passingNode: AxeNodeResult = {
        any: [],
        none: [],
        all: [],
        instanceId: 'id-pass',
        html: 'html-pass',
        target: ['passTarget1', 'passTarget2'],
    };

    const incompleteNode: AxeNodeResult = {
        any: [],
        none: [],
        all: [],
        instanceId: 'id-incomplete',
        html: 'html-incomplete',
        target: ['incompleteTarget1'],
    };

    const failedRules: RuleResult[] = [
        {
            id: 'id1',
            nodes: [node1, node2],
            description: 'des1',
            help: 'help1',
            guidanceLinks: [
                {
                    text: 'text1',
                    href: 'testurl1',
                },
            ],
            helpUrl: 'https://id1',
        },
        {
            id: 'id2',
            nodes: [node3],
            description: 'des2',
            help: 'help2',
            guidanceLinks: [
                {
                    text: 'text2',
                    href: 'testurl2',
                },
            ],
            helpUrl: 'https://id2',
        },
    ];

    function createTestResults(): ScanResults {
        return {
            passes: [
                {
                    id: 'test',
                    nodes: [passingNode],
                },
            ],
            violations: failedRules,
            targetPageTitle: '',
            targetPageUrl: '',
        } as ScanResults;
    }

    function createTestResultsWithIncompletes(): ScanResults {
        return {
            passes: [
                {
                    id: 'test',
                    nodes: [passingNode],
                },
            ],
            violations: failedRules,
            incomplete: [
                {
                    id: 'test2',
                    nodes: [incompleteNode],
                },
            ],
            targetPageTitle: '',
            targetPageUrl: '',
        } as ScanResults;
    }
});
