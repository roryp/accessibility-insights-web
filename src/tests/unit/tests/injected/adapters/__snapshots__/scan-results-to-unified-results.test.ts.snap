// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ScanResults to Unified Results Test conversion works fine when there is no data in scanresults 1`] = `Array []`;

exports[`ScanResults to Unified Results Test conversion works with filled up passes and failures value in scan results 1`] = `
Array [
  Object {
    "descriptors": Object {
      "snippet": "html1",
    },
    "identifiers": Object {
      "conciseName": "id1",
      "css-selector": "target1;id1",
      "identifier": "target1;id1",
    },
    "resolution": Object {
      "how-to-fix-web": Object {
        "all": Array [],
        "any": Array [
          "any-test-message",
        ],
        "none": Array [],
      },
      "howToFixSummary": "how to fix 1",
    },
    "ruleId": "id1",
    "status": "fail",
    "uid": "gguid-mock-stub",
  },
  Object {
    "descriptors": Object {
      "snippet": "html2",
    },
    "identifiers": Object {
      "conciseName": "id2",
      "css-selector": "target2;id2",
      "identifier": "target2;id2",
    },
    "resolution": Object {
      "how-to-fix-web": Object {
        "all": Array [],
        "any": Array [],
        "none": Array [
          "none-test-message",
        ],
      },
      "howToFixSummary": "how to fix 2",
    },
    "ruleId": "id1",
    "status": "fail",
    "uid": "gguid-mock-stub",
  },
  Object {
    "descriptors": Object {
      "snippet": "html3",
    },
    "identifiers": Object {
      "conciseName": "id3",
      "css-selector": "target3;id3",
      "identifier": "target3;id3",
    },
    "resolution": Object {
      "how-to-fix-web": Object {
        "all": Array [
          "all-test-message",
        ],
        "any": Array [],
        "none": Array [],
      },
      "howToFixSummary": "how to fix 3",
    },
    "ruleId": "id2",
    "status": "fail",
    "uid": "gguid-mock-stub",
  },
  Object {
    "descriptors": Object {
      "snippet": "html-pass",
    },
    "identifiers": Object {
      "conciseName": "passTarget2",
      "css-selector": "passTarget1;passTarget2",
      "identifier": "passTarget1;passTarget2",
    },
    "resolution": Object {
      "how-to-fix-web": Object {
        "all": Array [],
        "any": Array [],
        "none": Array [],
      },
      "howToFixSummary": undefined,
    },
    "ruleId": "test",
    "status": "pass",
    "uid": "gguid-mock-stub",
  },
]
`;

exports[`ScanResults to Unified Results Test needs review conversion works fine when there is no data in scanresults 1`] = `Array []`;

exports[`ScanResults to Unified Results Test needs review conversion works with filled up passes, failures and incomplete values in scan results 1`] = `
Array [
  Object {
    "descriptors": Object {
      "snippet": "html-incomplete",
    },
    "identifiers": Object {
      "conciseName": "incompleteTarget1",
      "css-selector": "incompleteTarget1",
      "identifier": "incompleteTarget1",
    },
    "resolution": Object {
      "how-to-fix-web": Object {
        "all": Array [],
        "any": Array [],
        "none": Array [],
      },
      "howToFixSummary": undefined,
    },
    "ruleId": "test2",
    "status": "unknown",
    "uid": "gguid-mock-stub",
  },
  Object {
    "descriptors": Object {
      "snippet": "html1",
    },
    "identifiers": Object {
      "conciseName": "id1",
      "css-selector": "target1;id1",
      "identifier": "target1;id1",
    },
    "resolution": Object {
      "how-to-fix-web": Object {
        "all": Array [],
        "any": Array [
          "any-test-message",
        ],
        "none": Array [],
      },
      "howToFixSummary": "how to fix 1",
    },
    "ruleId": "id1",
    "status": "fail",
    "uid": "gguid-mock-stub",
  },
  Object {
    "descriptors": Object {
      "snippet": "html2",
    },
    "identifiers": Object {
      "conciseName": "id2",
      "css-selector": "target2;id2",
      "identifier": "target2;id2",
    },
    "resolution": Object {
      "how-to-fix-web": Object {
        "all": Array [],
        "any": Array [],
        "none": Array [
          "none-test-message",
        ],
      },
      "howToFixSummary": "how to fix 2",
    },
    "ruleId": "id1",
    "status": "fail",
    "uid": "gguid-mock-stub",
  },
  Object {
    "descriptors": Object {
      "snippet": "html3",
    },
    "identifiers": Object {
      "conciseName": "id3",
      "css-selector": "target3;id3",
      "identifier": "target3;id3",
    },
    "resolution": Object {
      "how-to-fix-web": Object {
        "all": Array [
          "all-test-message",
        ],
        "any": Array [],
        "none": Array [],
      },
      "howToFixSummary": "how to fix 3",
    },
    "ruleId": "id2",
    "status": "fail",
    "uid": "gguid-mock-stub",
  },
]
`;
